### PR TITLE
Wiki doc

### LINKED ISSUE #639: out of date documentation
Documentation refers to snippets in `HelloRadium/minimalradium.cpp` which has been removed (commit 6df619d1b0dd49b069842b92f016494183d8c7df)
Some snippets are defined, but current doc on material is broken.
Please update ;)

### LINKED ISSUE #678: Wiki vs. doc
Some on the wiki pages should be in the documentation.
They may be actually deprecated right now, so they need an update when moving from wiki to doc.

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

Work plan
- [x]  create doc files with the content of wiki pages
- [x]  delete wiki pages
- [x] reorganize doc
- [ ] update copied wiki pages according to current version of radium
- [ ] also update deprecated doc fix #639

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Doc only, refactor, update, remove warnings.
Still wip, help welcome.

* **What is the current behavior?** (You can also link to an open issue here)
Fix #678

### CODE REVIEW

File: doc/basics/compilation.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ RADIUM_GENERATE_LIB_CORE:BOOL=ON
 // Include Radium::Engine in CMake project
 RADIUM_GENERATE_LIB_ENGINE:BOOL=ON
 --
-// Include Radium::GuiBase in CMake project
+// Include Radium::Gui in CMake project
```
Comment (nmellado) [inline doc/basics/compilation.md:73]: Shouldn’t we rename the variable as well, for consistency ?

File: doc/concepts/componentMessenger.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    if (ComponentMessenger::getInstance()->canRw<T>( entity, "name"))
+    {
+        T& data = ComponentMessenger::getInstance()->rw<T>( entity, "name" );
+    }
+
+Here a pointer of the data is basically shared and everyone has only to do:
+
+# The `setupIO()` function #
+
+A convention is to use the `setupIO()` function to interact with the messenger.
```
Comment (nmellado) [inline doc/concepts/componentMessenger.md:56]: I don’t understand what that means exactly ? What should I put in it ?

Comment (dlyr) [reply doc/concepts/componentMessenger.md:56]: Well this documentation is just a copy/paste of the wiki for the record. It's marked as deprecated and need rewriting.

File: doc/basics/compilation.md (hunk 1 above)
Comment (dlyr) [reply doc/basics/compilation.md:73]: Actually it's the case in CMakeLists.txt, need update here

File: doc/concepts/componentMessenger.md (hunk 2 above)
Comment (nmellado) [reply doc/concepts/componentMessenger.md:56]: Ok

File: doc/basics/compilation.md (hunk 1 above)
Comment (dlyr) [reply doc/basics/compilation.md]: (it was modified in another PR, I mess up in the pushes)

**Review by nmellado — APPROVED** [review]

**Pull request merged**